xl/playlist.py

```diff
@@ -297,34 +297,25 @@ def get_track_export_path(
         :param track_path: the path of the track
         :param options: options
         """
-        if options is not None and options.relative:
-            playlist_file = Gio.File.new_for_uri(playlist_path)
-            # Strip playlist filename from export path
-            export_path = playlist_file.get_parent().get_uri()
 
-            try:
-                export_path_components = urllib.parse.urlparse(export_path)
-                track_path_components = urllib.parse.urlparse(track_path)
-            except (AttributeError, ValueError):  # None, empty path
-                pass
-            else:
-                # Only try to retrieve relative paths for tracks with
-                # the same URI scheme and location as the playlist
-                if (
-                    export_path_components.scheme == track_path_components.scheme
-                    and export_path_components.netloc == track_path_components.netloc
-                ):
-                    # Gio.File.get_relative_path does not generate relative paths
-                    # for tracks located above the playlist in the path hierarchy,
-                    # thus process both paths as done here
-                    track_path = os.path.relpath(
-                        track_path_components.path, export_path_components.path
-                    )
+        track_path_components = urllib.parse.urlparse(track_path)
+
+        if not track_path_components.scheme == 'file':
+            # return remote files as is
+            return track_path
+
+        track_path = Gio.File.new_for_uri(track_path)
+        if options is None or not options.relative:
+            # return local files as is but as path instead of uri
+            return track_path.get_path()
+
+        # calculate relative path
+        playlist_file = Gio.File.new_for_uri(playlist_path)
+        export_path = playlist_file.get_parent().get_path()
+
+        track_path = os.path.relpath(track_path.get_path(), export_path)
```

Comment:
Hmm, what should happen if a relative path cannot be computed?

For example, on Windows, music files might be on one drive, and playlist is being stored on another drive. In such case, `os.path.relpath()` will raise an exception, and the file will end up being skipped again. Should such files be returned with absolute path instead?

The changes here remove support for using relative paths with non-local schema in case when both media files and the playlist use the same schema and location (for example, remote gvfs mounts under linux). I have no idea if this worked as it should have and if it is being used, though.

Reply:
I cannot confirm on Win11 with Exaile built in https://ci.appveyor.com/project/ExaileDevelopmentTeam/exaile/builds/52247370

With relative and with absolut paths I get `\` as separator.

@desbest Can you check also?

Reply:
Hmm, you're right. I guess this is because I was running from source in the msys environment.

The msys-installed python's `ntpath.py` has this block:

```python
if sys.platform == "win32" and os.environ.get("MSYSTEM", ""):
    sep = '/'
    altsep = '\\'
else:
    sep = '\\'
    altsep = '/'
```

So it ends up with `/` as separator when ran from source using msys python in msys environment, and `\` when ran as PyInstaller-frozen, because `MSYSTEM` is not set.